pkg/keys/ecdsa.go

```diff
@@ -73,14 +65,87 @@ func (p *p256Verifier) UnmarshalPublicKey(key *data.PublicKey) error {
 		return err
 	}
 
-	curve := elliptic.P256()
+	ecdsaKey, ok := p.PublicKey.PublicKey.(*ecdsa.PublicKey)
+	if !ok {
+		return fmt.Errorf("invalid public key")
+	}
+	p.ecdsaKey = ecdsaKey
+	p.key = key
+	return nil
+}
+
+type ecdsaSigner struct {
+	*ecdsa.PrivateKey
+}
+
+type ecdsaPrivateKeyValue struct {
+	Private string         `json:"private"`
+	Public  *PKIXPublicKey `json:"public"`
+}
 
-	// Parse as uncompressed marshalled point.
-	x, _ := elliptic.Unmarshal(curve, p.PublicKey)
-	if x == nil {
-		return errors.New("tuf: invalid ecdsa public key point")
+func (s *ecdsaSigner) PublicData() *data.PublicKey {
+	keyValBytes, _ := json.Marshal(EcdsaVerifier{PublicKey: &PKIXPublicKey{PublicKey: s.Public()}})
+	return &data.PublicKey{
```

Comment:
(BTW, side note: I'm beginning to think the TUF spec should also specify the encoding expect for public keys ahead of time, but that's a different story for another time.) @mnm678

Reply:
Agreed! There currently is consistency of repo key storage across impls.